deployment/ccip/changeset/deployer_group.go

```diff
@@ -94,77 +126,141 @@ func (d *DeployerGroup) GetDeployer(chain uint64) (*bind.TransactOpts, error) {
 		startingNonce = new(big.Int).SetUint64(nonce)
 	}
 
+	dc := d.deploymentContext
 	sim.Signer = func(a common.Address, t *types.Transaction) (*types.Transaction, error) {
+		txCount, err := d.getTransactionCount(chain)
+		if err != nil {
+			return nil, err
+		}
+
+		currentNone := big.NewInt(0).Add(startingNonce, txCount)
+
 		// Update the nonce to consider the transactions that have been sent
-		sim.Nonce = big.NewInt(0).Add(startingNonce, big.NewInt(int64(len(d.transactions[chain]))+1))
+		sim.Nonce = big.NewInt(0).Add(currentNone, big.NewInt(1))
 
 		tx, err := oldSigner(a, t)
 		if err != nil {
 			return nil, err
 		}
-		d.transactions[chain] = append(d.transactions[chain], tx)
+		dc.transactions[chain] = append(dc.transactions[chain], tx)
 		return tx, nil
 	}
 	return sim, nil
 }
 
-func (d *DeployerGroup) Enact(deploymentDescription string) (deployment.ChangesetOutput, error) {
+func (d *DeployerGroup) getContextChainInOrder() []*DeploymentContext {
+	contexts := make([]*DeploymentContext, 0)
+	for c := d.deploymentContext; c != nil; c = c.previousConfig {
+		contexts = append(contexts, c)
+	}
+	slices.Reverse(contexts)
+	return contexts
+}
+
+func (d *DeployerGroup) getTransactions() map[uint64][]*types.Transaction {
+	transactions := make(map[uint64][]*types.Transaction)
+	for _, c := range d.getContextChainInOrder() {
+		for k, v := range c.transactions {
+			transactions[k] = append(transactions[k], v...)
+		}
+	}
+	return transactions
+}
+
+func (d *DeployerGroup) getTransactionCount(chain uint64) (*big.Int, error) {
+	txs := d.getTransactions()
+	return big.NewInt(int64(len(txs[chain]))), nil
+}
+
+func (d *DeployerGroup) Enact() (deployment.ChangesetOutput, error) {
 	if d.mcmConfig != nil {
-		return d.enactMcms(deploymentDescription)
+		return d.enactMcms()
 	}
 
 	return d.enactDeployer()
 }
 
-func (d *DeployerGroup) enactMcms(deploymentDescription string) (deployment.ChangesetOutput, error) {
-	batches := make([]timelock.BatchChainOperation, 0)
-	for selector, txs := range d.transactions {
-		mcmOps := make([]mcms.Operation, len(txs))
-		for i, tx := range txs {
-			mcmOps[i] = mcms.Operation{
-				To:    *tx.To(),
-				Data:  tx.Data(),
-				Value: tx.Value(),
+func (d *DeployerGroup) enactMcms() (deployment.ChangesetOutput, error) {
+	contexts := d.getContextChainInOrder()
+	proposals := make([]timelock.MCMSWithTimelockProposal, 0)
+	for _, dc := range contexts {
+		batches := make([]timelock.BatchChainOperation, 0)
+		for selector, txs := range dc.transactions {
+			mcmOps := make([]mcms.Operation, len(txs))
+			for i, tx := range txs {
+				mcmOps[i] = mcms.Operation{
+					To:    *tx.To(),
+					Data:  tx.Data(),
+					Value: tx.Value(),
+				}
 			}
+			batches = append(batches, timelock.BatchChainOperation{
+				ChainIdentifier: mcms.ChainIdentifier(selector),
+				Batch:           mcmOps,
```

Comment:
If batch length is 0, BuildProposalFromBatches will return an error. We should exit gracefully if there are no transactions.

Reply:
& add test case for this